docs/architecture/how/gas.md

```diff
@@ -20,7 +20,7 @@ The topic is split into several sections.
 2. [Gas Price](#gas-price):
     - [Block-Level Gas Price](#block-level-gas-price): How the block-level gas price is determined.
     - [Pessimistic Gas Price](#pessimistic-gas-price): How worst-case gas pricing is estimated.
-    - [Effective Gas Purchase Cost](#effective-gas-purchase-cost): The cost paid for a receipt.
+    - [Gas Refund Tax](#gas-refund-tax): The cost paid for a receipt.
```

Comment:
kind of nitpicky but I would not call it a "tax" or "penalty", "Gas refund fee" sounds best to me.

Reply:
Hm, the NEP authors specifically called it a penalty. Empathizing that it's not just a regular fee covering chunk space but rather a disincentive to not attach too much gas.

Personally, I don't like "penalty" too much, either. But I've already used it in the code implementing all of this, including the config options here: https://github.com/near/nearcore/blob/f381b9dbf0c8381892c821341ed6fe62a8125502/core/parameters/res/runtime_configs/78.yaml#L10-L20

We can still go with "gas refund fee" for the documentation here. Then I can also use this terminology for the contract developer docs and the nomicon, keeping it persistent. But I think in the nearcore development guide, I should at least mention in a side comment that we call it "penalty" in the code and why.